src/event-bridge/event-bus.ts

```diff
@@ -116,8 +132,33 @@ export interface IEventBus<E extends EventBusRuleInput = EventBusRuleInput>
     event: EventBusPutEventInput<E>,
     ...events: EventBusPutEventInput<E>[]
   ): void;
+
+  /**
+   * Creates a rule that matches all events on the bus.
+   *
+   * When no scope or id are given, the bus is used as the scope and the id will be `all`.
+   * The rule created will be a singleton.
+   *
+   * When scope and id are given, a new rule will be created each time.
+   *
+   * Like all functionless, a rule is only created when the rule is used with `.pipe` or the rule is retrieved using `.rule`.
+   *
+   * ```ts
+   * const bus = new EventBus(scope, 'bus');
+   * const func = new Function(scope, 'func', async (payload: {id: string}) => console.log(payload.id));
+   *
+   * bus
+   *  .all()
+   *  .map(event => ({id: event.id}))
+   *  .pipe(func);
+   * ```
+   */
+  all(): EventBusPredicateRuleBase<E>;
```

Comment:
Are we going to rename EventBusPredicateRuleBase also? PredicateRuleBase?

Reply:
Sure